flutter_example/lib/macros/auto_listenable.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:macro_builder/definition.dart';
+import 'package:collection/collection.dart';
+
+const autoListenable = AutoListenable();
+
+class AutoListenable implements ClassDeclarationMacro, MethodDefinitionMacro {
+  const AutoListenable();
+
+  /// In this phase we add overrides to all desired methods and apply ourself
+  /// to them.
+  @override
+  void visitClassDeclaration(
+      ClassDeclaration declaration, ClassDeclarationBuilder builder) {
+    // TODO: https://github.com/jakemac53/macro_prototype/issues/33
+    if (!declaration.hackyIsSubtypeOf('State')) {
+      throw ArgumentError('@autoListenable can only be used on State classes');
+    }
+    var widgetClass = declaration.widgetClass;
+
+    if (declaration.initState == null) {
+      builder.addToClass(Declaration('''
+@override
+@autoListenable
+external void initState();
+'''));
+    } else {
+      // https://github.com/jakemac53/macro_prototype/issues/34
+      throw ArgumentError(
+          '@autoListenable isn\'t compatible with a custom `initState` '
```

Comment:
That's a bummer 😄

Reply:
Ya, just something specific to this prototype though. If I have the time I will try and make it work better here haha but this is all just throwaway stuff so I don't think its critical I get all these cases worked out.